3.workload.md

```diff
@@ -0,0 +1,99 @@
+# 3. Workload
+
+This section defines the workload definition schematics.
+
+Usually, workload definitions are provided by the OAM runtime (just as traits are) so that application developers may inspect the OAM runtime and learn what workloads are available for use.  
+
+## Top-Level Attributes
+
+The following attributes are common across all schemata defined in this document. They follow the [Kubernetes specification](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#required-fields).
+
+Here are the attributes that provide top-level information about the workload definition.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `apiVersion` | `string` | Y | | The specific version of the specification in use. The core types use `core.oam.dev/v1alpha2` |
+| `kind` | `string` | Y || The string `WorkloadDefinition` |
+| `metadata` | [`Metadata`](#metadata) | Y | | WorkloadDefinition metadata. |
+| `spec`| [`Spec`](#spec) | Y | | A container for the workload definition. |
+
+### Metadata
+
+The metadata describes this workload definition. The `metadata` section is defined in [the meta section of part 2](2.overview_and_terminology.md#Metadata)
+
+### Spec
+
+The specification section contains a reference to the workload definition.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `definitionRef` | [`DefinitionRef`](#DefinitionRef) | Y | | Workload schema reference. |
+
+
+#### DefinitionRef
+
+DefinitionRef defines an index to find the workload schema.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `schemaName` | `string` | Y | | The workload schema's name. |
+
+The workload schema definition itself __MUST__ contain information that can be used to uniquely identify it. OAM runtime will use that information in a [Component](4.component.md) to refer to the workload schema instead of using the name of the workload definition.
+ We recommend to use the format [Group/Version/Kind](2.overview_and_terminology.md#Group) described in section 2 as the unique identification in the schema. 
+
+We recommend that the `schemaName` here follows the format described in [the Representations of Group/Version/Kind in section 2](2.overview_and_terminology.md#Representations).
+
+We also recommend that the `name` of the workloadDefinition to be the same as the `schemaName` it refers to. Here is an example
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: WorkloadDefinition
+metadata:
+  name: schema.example.oam.dev
+spec:
+  definitionRef:
+    schemaName: schema.example.oam.dev
+```
```

Comment:
the example here doesn't follow the recommendation we just gave above to use GVK. Is it enough to simply recommend GVK, or should it be required?

Reply:
We didn't recommend GVK as the result of several discussions says no "V" needed. I will modify the wording a bit.

Reply:
Hi @vturecek 

Please see referenced discussions 
- https://github.com/oam-dev/spec/pull/316#discussion_r383492681
- https://github.com/crossplane/crossplane/pull/1276#discussion_r382321173 (@negz 's PR in crossplane)